### PR TITLE
Update docs for cakephp/cakephp#12486

### PR DESCRIPTION
**Pull request by markstory**
Update docs for cakephp/cakephp#12486

cc @chinpei215

### CODE REVIEW

File: en/core-libraries/validation.rst [hunk 1]
```diff
@@ -96,29 +96,44 @@ If you have multiple fields that are required, you can define them as a list::
...
+#. ``allowEmptyDate()`` Should be used when you want to accept an empty string,
+   or an array that is marshalled into a time field.
+#. ``allowEmptyDateTime()`` Should be used when you want to accept an empty
+   string or an array that is marshalled into a datetime or timestamp field.
+#. ``allowEmptyFile()`` Should be used when you want to accept an array that
+   is contains an empty uploaded file.
+
+You can also use ``notEmpty()`` to mark a field invalid if any 'empty' value is
+used. In general, it is recommended that you do not use ``notEmpty()`` and use more
+specific validators instead. practice to use more specific validators.
```
Comment (ravage84) [inline en/core-libraries/validation.rst]: > practice to use more specific validators.

Is this an incomplete sentence?

File: en/core-libraries/validation.rst [hunk 2]
```diff
@@ -36,7 +36,7 @@ validate::
                 'message' => 'Titles need to be at least 10 characters long',
             ]
         ])
-        ->allowEmpty('published')
+        ->allowEmptyString('published')
```
Comment (ravage84) [inline en/core-libraries/validation.rst]: Why not ``allowEmptyDateTime()``?

File: en/core-libraries/validation.rst [hunk 3]
```diff
@@ -96,29 +96,44 @@ If you have multiple fields that are required, you can define them as a list::
...
+#. ``allowEmptyDateTime()`` Should be used when you want to accept an empty
+   string or an array that is marshalled into a datetime or timestamp field.
+#. ``allowEmptyFile()`` Should be used when you want to accept an array that
+   is contains an empty uploaded file.
+
+You can also use ``notEmpty()`` to mark a field invalid if any 'empty' value is
+used. In general, it is recommended that you do not use ``notEmpty()`` and use more
+specific validators instead. practice to use more specific validators.
+
+The ``allowEmpty`` methods support a mode parameter that allows you to control
```
Comment (ravage84) [inline en/core-libraries/validation.rst]: ``allowEmptyTYPE`` or  ``allowEmptyX`` or ``allowEmpty*``?

File: en/core-libraries/validation.rst [hunk 4]
```diff
@@ -96,29 +96,44 @@ If you have multiple fields that are required, you can define them as a list::
...
-empty, the values ``''``, ``null``, ``false``, ``[]``, ``0``, ``'0'`` are
-accepted.
-
 An example of these methods in action is::
 
-    $validator->allowEmpty('published')
-        ->notEmpty('title', 'Title cannot be empty')
-        ->notEmpty('body', 'Body cannot be empty', 'create')
-        ->allowEmpty('header_image', 'update');
+    // Prior to 3.6.12 you should use allowEmpty().
```
Comment (ravage84) [inline en/core-libraries/validation.rst]: Actually, you "must"...

File: en/core-libraries/validation.rst [hunk 5]
```diff
@@ -96,29 +96,44 @@ If you have multiple fields that are required, you can define them as a list::
...
+    $validator->allowEmptyString('published')
```
Comment (ravage84) [inline en/core-libraries/validation.rst]: Why not ``allowEmptyDateTime()``?

File: en/core-libraries/validation.rst [hunk 6]
```diff
@@ -96,29 +96,44 @@ If you have multiple fields that are required, you can define them as a list::
...
+which empty values are accepted and not forwarded to other validation rules for
+the named field. CakePHP provides empty value support for five different shapes
+of data:
+
+#. ``allowEmptyString()`` Should be used when you want to only accept
+   an empty string.
+#. ``allowEmptyArray()`` Should be used when you want to accept an array.
+#. ``allowEmptyDate()`` Should be used when you want to accept an empty string,
+   or an array that is marshalled into a date field.
+#. ``allowEmptyDate()`` Should be used when you want to accept an empty string,
```
Comment (chinpei215) [inline en/core-libraries/validation.rst]: allowEmptyTime

**Comment by markstory** [comment]
@chinpei215 @ravage84 Thank you for the review. I've made the recommended updates.

**Pull request merged**